src/desktop_notifier/backends/winrt.py

```diff
@@ -225,25 +225,21 @@ def _on_activated(
         if not sender:
             return
 
-        notification = self._clear_notification_from_cache(sender.tag)
-
         if not boxed_activated_args:
             return
 
         activated_args = ToastActivatedEventArgs._from(boxed_activated_args)
         action_id = activated_args.arguments
 
         if action_id == DEFAULT_ACTION:
-            self.handle_clicked(sender.tag, notification)
-
+            self.handle_clicked(sender.tag)
         elif action_id == REPLY_ACTION and activated_args.user_input:
             boxed_reply = activated_args.user_input[REPLY_TEXTBOX_NAME]
             reply = unbox(boxed_reply)
-            self.handle_replied(sender.tag, reply, notification)
-
+            self.handle_replied(sender.tag, reply)
         elif action_id.startswith(BUTTON_ACTION_PREFIX):
             button_id = action_id.replace(BUTTON_ACTION_PREFIX, "")
-            self.handle_button(sender.tag, button_id, notification)
+            self.handle_button(sender.tag, button_id)
```

Comment:
In case of a fall-through, we will no longer clear the notification from the cache here. I'm not sufficiently familiar with Windows Toast Notifications to know if this is possible, but let's make sure that the notification is always removed from our cache after close.

Reply:
The notification is removed from the cache within `handle_button()`.

Reply:
Ah, that's what you mean. I'm no expert of that API either, but I think your assessment, that this case is ultimately unreachable, is correct. But a safe-guard is a good idea nevertheless. However, I think that we shouldn't just clear the cache, but rather trigger an event. I think `on_clicked` (i.e. moving `action_id == DEFAULT_ACTION` to else) would be most suitable. WDYT?